ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/TransactionReaper.java

```diff
@@ -137,15 +138,19 @@ public final void check()
                     nextDynamicCheckTime.set(Long.MAX_VALUE);
                     return;
                 } else {
-                    final long nextTimeout = reaperElement.getAbsoluteTimeout();
-                    if(nextTimeout > now) {
-                        nextDynamicCheckTime.set(nextTimeout);
+                    final long nextCheck = reaperElement.getNextCheckAbsoluteMillis();
+                    if(nextCheck > now) {
+                        nextDynamicCheckTime.set(nextCheck);
                         return; // nothing to do yet.
                     }
                 }
             }
 
-            tsLogger.i18NLogger.warn_coordinator_TransactionReaper_18(reaperElement._control.get_uid(), reaperElement.statusName());
+            if(reaperElement._status != ReaperElement.RUN ||
```

Comment:
I cannot follow the logic path that stops the warning from being printed when reaperElement._status is ReaperElement.TRACE. @jhalliday will you clarify this for me?

Reply:
It's never TRACE at that point. It's RUN, until the reaper sets it to TRACE below, based on the same comparison of the timestamps. If TRACE was used in the way you think, there would need to be a mechanism for the worker thread to wake the reaper thread so that e.g. TRACE_DONE could be reset to RUN/TRACE. That doesn't exist as present - the wake is always time based, never completion based. We could do it be having an interval within which the worker is expected to complete the tracing, much like we have an interval in which it's expected to complete the cancel, but that seems like overkill.

Reply:
OK, thanks for the explanation. Let's hope I don't have to change this code very often when you're not around.